lib/poll.rb

```diff
@@ -2,19 +2,27 @@ class Poll
   class InvalidCandidateError < StandardError
   end
 
-  attr_reader :title, :candidates, :votes
+  class InvalidVoteTimeError < StandardError
+  end
+
+  attr_reader :title, :candidates, :votes, :deadline
 
-  def initialize(title, candidates)
+  def initialize(title, candidates, deadline = nil)
     @title = title
     @candidates = candidates
     @votes = []
+    @deadline = deadline
   end
 
   def add_vote(vote)
     unless candidates.include?(vote.candidate)
       raise InvalidCandidateError, "Candidate '#{vote.candidate}' is invalid"
     end
 
+    if deadline != nil && Time.now > deadline
```

Comment:
[nit] 細かいですが Ruby には [`Object#nil?`](https://docs.ruby-lang.org/ja/latest/method/Object/i/nil=3f.html) というメソッドがあり、`nil` と比較するよりはこちらを使うのが一般的です。

```suggestion
    if !deadline.nil? && Time.now > deadline
```

あるいは、nil (と false) が偽、それ以外が真になることを利用して

```suggestion
    if deadline && Time.now > deadline
```

と書くのも Ruby っぽい書き方かなと思います。

※ [nit] は nitpick の略で「細かいのでどっちでもいいけどちょっと気になる」みたいなニュアンスの時に使います。
参考: https://techlife.cookpad.com/entry/2018/06/19/110000